lib/cfg-lexer.c

```diff
@@ -1188,6 +1188,7 @@ static const gchar *lexer_contexts[] =
   [LL_CONTEXT_CLIENT_PROTO] = "client-proto",
   [LL_CONTEXT_SERVER_PROTO] = "server-proto",
   [LL_CONTEXT_OPTIONS] = "options",
+  [LL_CONTEXT_HTTP_AUTH_HEADER] = "http-auth-header",
 };
```

Comment:
This is pluggable? We could perhaps reuse the context_inner_destination context and introduce an api between the plugins.

Reply:
Why this is a big issue?

Reply:
@bazsi: yes, it can be done by using the `LL_CONTEXT_INNER_DEST`, but I guess we need the very same to the http source (and yes, I know that there is `LL_CONTEXT_INNER_SRC`).
So what you really want is to remove the `LL_CONTEXT_HTTP_AUTH_HEADER` and 
use the `LL_CONTEXT_INNER_DEST` instead? Is that all? Any other issue?

Reply:
@bazsi: hmmm... after we discussed it with @MrAnno, I think -as it is a requirement to make it possible to implement auth-headers as plugins(including PE-only)- using inner_context is  not a good idea.
On the other hand, I understand your hesitation on 'polluting' the 'context-space', but I think it can be changed/fixed in the future without breaking the config syntax, so I mark your change request as resolved.

Reply:
Adding the `http-auth` interface to the core lib looks a bit weird, but the current licensing forced us to do such thing AFAIK.

However, I think reusing the inner destination context is not a good idea. For `hook-commands` and `disk-buffer`, it works perfectly, they don't have any "communication interface" with the specific driver. In case of `HttpAuthHeader`, `cfg_parse_plugin()` returns a instance of this interface, so we assume a `HttpAuthHeader` instance here.

If we changed the context to inner destination, it would allow the user to specify a `disk-buffer()` or `hook-commands()` plugin in `auth-header()`. With the current implementation, it results in a crash.